### ISSUE TITLE
DHCP/Host-Only Ethernet Adapter error on `fin vm start`

### ISSUE THREAD

**Issue by RoloDMonkey** [docksal/docksal#198]
**Description**

I am trying to install Docksal on a Windows 7 machine. I get the error below during `fin vm start`.  I am on a VPN, and this is probably related to issue #117.

**Describe the results you received:**

```
{ ~ }  » fin vm start                                                                                                                                         ~ 1
Creating docker machine 'docksal'...
Running pre-create checks...
Creating machine...
(docksal) Downloading C:\Users\Clayton.Predmore\.docker\machine\cache\boot2docker.iso from https://github.com/boot2docker/boot2docker/releases/download/v17.04.0-ce/boot2docker.iso...
(docksal) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(docksal) Creating VirtualBox VM...
(docksal) Creating SSH key...
(docksal) Starting the VM...
(docksal) Check network to re-create if needed...
(docksal) Windows might ask for the permission to create a network adapter. Sometimes, such confirmation window is minimized in the taskbar.
(docksal) Found a new host-only adapter: "VirtualBox Host-Only Ethernet Adapter"
(docksal) Windows might ask for the permission to configure a network adapter. Sometimes, such confirmation window is minimized in the taskbar.
(docksal) Windows might ask for the permission to configure a dhcp server. Sometimes, such confirmation window is minimized in the taskbar.
Error creating machine: Error in driver during machine creation: Unable to start the VM: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe startvm docksal --type headless failed:
VBoxManage.exe: error: Failed to open/create the internal network 'HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter' (VERR_INTNET_FLT_IF_NOT_FOUND).
VBoxManage.exe: error: Failed to attach the network LUN (VERR_INTNET_FLT_IF_NOT_FOUND)
VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component ConsoleWrap, interface IConsole

Details: 00:00:01.474410 Power up failed (vrc=VERR_INTNET_FLT_IF_NOT_FOUND, rc=E_FAIL (0X80004005))
 ERROR:  Proper creation of virtual machine has failed
         For details please refer to the log above.
         It is recommended to remove malfunctioned virtual machine.
Remove docksal? [y/n]:
```

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
fin sysinfo                                                                          ~ 1
███  OS & BASICS
CYGWIN_NT-6.1-WOW OCL0142874 1.7.35(0.287/5/3) 2015-03-04 12:07 i686 Cygwin
 Microsoft Windows [Version 6.1.7601]
fin version: 1.6.0
Mode:  VirtualBox VM
███  INSTANCES
NAME      ACTIVE   DRIVER       STATE     URL   SWARM   DOCKER    ERRORS
docksal   -        virtualbox   Stopped                 Unknown
███  DOCKER
DOCKER_HOST:

error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.28/version: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
Docker: Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 23:33:17 2017
 OS/Arch:      windows/amd64
███  DOCKER COMPOSE
Docker Compose: docker-compose version 1.12.0, build ee0f34e1
docker-py version: 2.2.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
███  DOCKER MACHINE
docker-machine.exe version 0.10.0, build 76ed2a6
███  VIRTUALBOX
5.1.22r115126
███  NETWORK INTERFACES
Name:            VirtualBox Host-Only Ethernet Adapter
GUID:            b4e44313-943f-4052-a57d-ab566f84112c
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     fe80:0000:0000:0000:cd7f:3ec0:b30d:30d2
IPV6NetworkMaskPrefixLength: 64
HardwareAddress: 0a:00:27:00:00:16
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter
  ```

</details>

**Comment by achekulaev** [comment]
Looks like a known [Virtualbox bug](https://www.virtualbox.org/ticket/14832) that happens for some.
People say this works for them: http://stackoverflow.com/a/33733454/1359178

**Comment by lmakarov** [comment]
http://stackoverflow.com/a/33733454/1359178 ring a bell for me.
Last time I had to deal with this issue the Windows box had to be reinstalled to get it resolved...
Hopefully there will be an easy way here.

**Comment by RoloDMonkey** [comment]
So, I re-installed VirtualBox with the NDIS5 option: http://stackoverflow.com/a/33733486

That got me beyond the error above, but now I am getting this:

```
...

(docksal) Windows might ask for the permission to configure a dhcp server. Sometimes, such confirmation window is minimized in the taskbar.
(docksal) Waiting for an IP...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with boot2docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...

This machine has been allocated an IP address, but Docker Machine could not
reach it successfully.

SSH for the machine should still work, but connecting to exposed ports, such as
the Docker daemon port (usually <ip>:2376), may not work properly.

You may need to add the route manually, or use another related workaround.

This could be due to a VPN, proxy, or host file configuration issue.

You also might want to clear any VirtualBox host only interfaces you are not using.
Checking connection to Docker...
Error creating machine: Error checking the host: Error checking and/or regenerating the certs: There was an error validating certificates for host "192.168.64.100:2376": dial tcp 192.168.64.100:2376: i/o timeout
You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'.
Be advised that this will trigger a Docker daemon restart which might stop running containers.
```

**Comment by achekulaev** [comment]
@RoloDMonkey could you run and attach sysinfo once again?

**Comment by RoloDMonkey** [comment]
```
███  OS & BASICS
CYGWIN_NT-6.1-WOW OCL0142874 1.7.35(0.287/5/3) 2015-03-04 12:07 i686 Cygwin
 Microsoft Windows [Version 6.1.7601]
fin version: 1.6.0
Mode:  VirtualBox VM
███  INSTANCES
NAME      ACTIVE   DRIVER       STATE     URL   SWARM   DOCKER   ERRORS
docksal            virtualbox   Timeout
███  DOCKER
DOCKER_HOST:

error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.28/version: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
Docker: Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 23:33:17 2017
 OS/Arch:      windows/amd64
███  DOCKER COMPOSE
Docker Compose: docker-compose version 1.12.0, build ee0f34e1
docker-py version: 2.2.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
███  DOCKER MACHINE
docker-machine.exe version 0.10.0, build 76ed2a6
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.28/info: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
███  VIRTUALBOX
5.1.22r115126
███  NETWORK INTERFACES
Name:            VirtualBox Host-Only Ethernet Adapter
GUID:            634e7f4a-fccb-4f7d-a46f-3845c1567653
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     fe80:0000:0000:0000:a460:7650:81d7:5967
IPV6NetworkMaskPrefixLength: 64
HardwareAddress: 08:00:27:00:6c:93
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter

███  MOUNTS
tmpfs on / type tmpfs (rw,relatime,size=917692k)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)
tmpfs on /dev/shm type tmpfs (rw,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
/dev/sda1 on /mnt/sda1 type ext4 (rw,relatime,data=ordered)
cgroup on /sys/fs/cgroup type tmpfs (rw,relatime,mode=755)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,relatime,cpuset)
cgroup on /sys/fs/cgroup/cpu type cgroup (rw,relatime,cpu)
cgroup on /sys/fs/cgroup/cpuacct type cgroup (rw,relatime,cpuacct)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,relatime,blkio)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,relatime,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,relatime,devices)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls type cgroup (rw,relatime,net_cls)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,relatime,perf_event)
cgroup on /sys/fs/cgroup/net_prio type cgroup (rw,relatime,net_prio)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,relatime,hugetlb)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,relatime,pids)
/dev/sda1 on /mnt/sda1/var/lib/docker/aufs type ext4 (rw,relatime,data=ordered)

```

**Comment by achekulaev** [comment]
Sometimes VirtualBox re-installation is not enough by itself and you need to restart your Windows. Could you try restarting Windows, removing VM with `fin vm remove` and start it again?

**Comment by RoloDMonkey** [comment]
I'm pretty sure the problem I am having is because of the VPN I am on.  I think 192.168... is already being used.

**Comment by achekulaev** [comment]
While it's very possible could you still try restarting Windows? As I've seen error similar to the last one you have during testing Docksal on Windows 7 in Vmware. VirtualBox could not use adapter properly until I restarted the host itself

**Comment by RoloDMonkey** [comment]
Restarting the Windows machine, and then doing a `fin vm remove`, `fin vm start` produced the same error.

**Comment by RoloDMonkey** [comment]
Could this be a permissions issue?  When I do `fin vm stop`, `fin vm start`, I get this.

```
fin vm start                                                                           ~
Starting "docksal"...
(docksal) Check network to re-create if needed...
(docksal) Waiting for an IP...
Machine "docksal" was started.
Waiting for SSH to be available...
Detecting the provisioner...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
Error checking TLS connection: Error checking and/or regenerating the certs: There was an error validating certificates for host "192.168.64.100:2376": dial tcp 192.168.64.100:2376: i/o timeout
You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'.
Be advised that this will trigger a Docker daemon restart which might stop running containers.

Configuring SMB shares...
Enter your Windows account password:
Adding docker SMB share...
mount: mounting //192.168.64.1/c on /c failed: Operation now in progress
exit status 255
Error creating share
```

**Comment by achekulaev** [comment]
Can you ping these IP addresses?

`ping 192.168.64.1`

`ping 192.168.64.100`

**Comment by RoloDMonkey** [comment]
```
 ping 192.168.64.1                                                                    ~ 1

Pinging 192.168.64.1 with 32 bytes of data:
General failure.
General failure.
General failure.
General failure.

Ping statistics for 192.168.64.1:
    Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),

```

```
 ping 192.168.64.100                                                                  ~ 1

Pinging 192.168.64.100 with 32 bytes of data:
Reply from 10.0.5.1: TTL expired in transit.
Reply from 10.0.5.1: TTL expired in transit.
Reply from 10.0.5.1: TTL expired in transit.
Reply from 10.0.5.1: TTL expired in transit.

Ping statistics for 192.168.64.100:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),

```

**Comment by achekulaev** [comment]
Yeah it VPN messing with your configuration. Basically routing is broken. Your network packets do not go to the VM. They go elsewhere.
https://superuser.com/a/659832

I'm not sure if I have enough knowledge to help out in this situation.

**Comment by lmakarov** [comment]
@RoloDMonkey What do you get if you disconnect from the VPN and reinitialize the vm?

```
fin vm rm
fin vm start
```

**Comment by RoloDMonkey** [comment]
That seemed to get me a lot further along. But, I am still getting the error when the script tries to mount the SMB share:

```
███  OS & BASICS
CYGWIN_NT-6.1-WOW OCL0142874 1.7.35(0.287/5/3) 2015-03-04 12:07 i686 Cygwin
 Microsoft Windows [Version 6.1.7601]
fin version: 1.6.0
Mode:  VirtualBox VM
███  INSTANCES
NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v17.04.0-ce
███  DOCKER
DOCKER_HOST:    tcp://192.168.64.100:2376

Docker: Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 23:33:17 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 18:45:47 2017
 OS/Arch:      linux/amd64
 Experimental: false
███  DOCKER COMPOSE
Docker Compose: docker-compose version 1.12.0, build ee0f34e1
docker-py version: 2.2.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
███  DOCKER MACHINE
docker-machine.exe version 0.10.0, build 76ed2a6
███  DOCKER: IMAGES
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
docksal/vhost-proxy   1.0                 2c8a8b1e62c4        2 months ago        164MB
docksal/dns           1.0                 20c7a535479d        3 months ago        61.6MB
docksal/ssh-agent     1.0                 148220cc033d        3 months ago        12.2MB
███  DOCKER: CONTAINERS
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS              PORTS                                      NAMES
e1a478320561        docksal/ssh-agent:1.0     "/run.sh ssh-agent"      9 minutes ago       Up 9 minutes                                                   docksal-ssh-agent
163422bc310a        docksal/dns:1.0           "/opt/entrypoint.s..."   9 minutes ago       Up 9 minutes        192.168.64.100:53->53/udp                  docksal-dns
d72600ca9ad9        docksal/vhost-proxy:1.0   "/usr/local/bin/st..."   9 minutes ago       Up 9 minutes        0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   docksal-vhost-proxy
███  VIRTUALBOX
5.1.22r115126
███  NETWORK INTERFACES
Name:            VirtualBox Host-Only Ethernet Adapter
GUID:            a3081def-b50d-4eca-aa7f-73e932009668
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     fe80:0000:0000:0000:00ea:c8a5:18c1:49db
IPV6NetworkMaskPrefixLength: 64
HardwareAddress: 08:00:27:00:90:3e
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter

███  MOUNTS
tmpfs on / type tmpfs (rw,relatime,size=917692k)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)
tmpfs on /dev/shm type tmpfs (rw,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
/dev/sda1 on /mnt/sda1 type ext4 (rw,relatime,data=ordered)
cgroup on /sys/fs/cgroup type tmpfs (rw,relatime,mode=755)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,relatime,cpuset)
cgroup on /sys/fs/cgroup/cpu type cgroup (rw,relatime,cpu)
cgroup on /sys/fs/cgroup/cpuacct type cgroup (rw,relatime,cpuacct)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,relatime,blkio)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,relatime,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,relatime,devices)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls type cgroup (rw,relatime,net_cls)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,relatime,perf_event)
cgroup on /sys/fs/cgroup/net_prio type cgroup (rw,relatime,net_prio)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,relatime,hugetlb)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,relatime,pids)
/dev/sda1 on /mnt/sda1/var/lib/docker/aufs type ext4 (rw,relatime,data=ordered)
none on /mnt/sda1/var/lib/docker/aufs/mnt/98ded954f071ed7156a5b117c45c5a5985b0ed980043a3818998f02b4d15a484 type aufs (rw,relatime,si=413d9cb6382a7a0a,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/d72600ca9ad96646fb9a113658782c635ccc22e9a279480516dee9583cf70058/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/136732baa25c type nsfs (rw)
none on /mnt/sda1/var/lib/docker/aufs/mnt/f3e2f0fad66549ff49754f1149d417546e121ca598427988e83a612f7c4fde1a type aufs (rw,relatime,si=413d9cb63b74ba0a,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/163422bc310a3b042acc3711755be34e707712faf24f4d459c9e322fbe96810c/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/ae74d05b5d4e type nsfs (rw)
none on /mnt/sda1/var/lib/docker/aufs/mnt/355685e45dbbf4f1fab4f81a914000f47fb0b50fb54ca636d29186a846227f94 type aufs (rw,relatime,si=413d9cb603b44a0a,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/e1a478320561fc5a22d2d9b9fbd45b6d1f31d650b918710f5c7463be2d192738/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/f67a8e894ee4 type nsfs (rw)
```

**Comment by achekulaev** [comment]
> I am still getting the error when the script tries to mount the SMB share
You forgot to share the output with the error you're getting

**Comment by RoloDMonkey** [comment]
https://github.com/docksal/docksal/issues/198#issuecomment-299008652

**Comment by achekulaev** [comment]
Please double check the error you're getting with VPN turned off. Because it can not be the same error as in https://github.com/docksal/docksal/issues/198#issuecomment-299008652

From your latest `sysinfo` output I see that Docker is reachable:
```
Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 18:45:47 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

While in that comment it says `dial tcp 192.168.64.100:2376: i/o timeout`. With error like that Docker would not be reachable.

**Comment by RoloDMonkey** [comment]
This is the error I get when the VPN is off.

```
{ ~ }  » fin vm start                                                                           ~
Starting "docksal"...
(docksal) Check network to re-create if needed...
(docksal) Waiting for an IP...
Machine "docksal" was started.
Waiting for SSH to be available...
Detecting the provisioner...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
Configuring SMB shares...
Enter your Windows account password:
mount: mounting //192.168.64.1/c on /c failed: Permission denied
exit status 255
Error creating share
```
However, it is kind of a moot point. For my client, it won't be acceptable to work outside their VPN.  I will check with a co-worker tomorrow and see if their computer also has the same problem with IP addresses, or if it is something I messed up the last time I was experimenting with VM settings.

**Comment by achekulaev** [comment]
Looks like VPN is a hell of an issue https://github.com/docker/toolbox/issues/346

**Comment by lmakarov** [comment]
Looks like VPN can easily break the host-only adapter routing that VirtualBox (or any hypervisor for that matter) sets up:
https://forums.virtualbox.org/viewtopic.php?f=6&t=61882

It also very much depends on the way the VPN client/policy is configured. Presumably, those more strict ones will take over the entire routing table and make the VM inaccessible from the host.

I have some hopes that using Docker for Mac/Windows would not suffer from these issues, since networking is done differently there. All ports published by Docker end up being mapped to your localhost (127.0.0.1), which makes it feel like it's your own machine listening on those ports.
I would be very surprised if a VPN client would be able to mess with that in any way.

@RoloDMonkey FYI Docker for Windows requires Windows 10 Pro/Ent. It is unfortunately not available for any Windows version below that. That's also the key reason we are not suggesting or supporting it as the mainstream solution (yet).

**Comment by achekulaev** [comment]
It might have been addressed in latest release 1.3.1. I would recommend updating, and restarting **host** workstation after the update.
